app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -203,35 +178,108 @@ class FirebaseAuthRepository : AuthRepository {
   private fun reauthenticate(
       user: FirebaseUser,
       password: String,
-      activity: Activity,
+      context: Context,
+      coroutineScope: CoroutineScope,
       onSuccess: () -> Unit,
       onFailure: () -> Unit
   ) {
     val isEmail = isEmailProvider(user)
 
     if (isEmail) {
-      val email = user.email ?: throw Exception("User email is null")
-      val credential: AuthCredential = EmailAuthProvider.getCredential(email, password)
-      user.reauthenticate(credential).addOnCompleteListener { task ->
-        if (task.isSuccessful) {
-          Log.d("DeleteAccount", "User re-authenticated")
-          onSuccess()
-        } else {
-          onFailure()
-        }
-      }
+      reauthenticateWithEmailAndPassword(user, password, onSuccess, onFailure)
     } else {
-      val googleProvider = OAuthProvider.newBuilder(GoogleAuthProvider.PROVIDER_ID)
-      user
-          .startActivityForReauthenticateWithProvider(activity, googleProvider.build())
-          .addOnFailureListener { e ->
-            Log.e("DeleteAccount", "Re-authentication failed: ${e.message}")
-            onFailure()
-          }
-          .addOnSuccessListener { result ->
-            Log.d("DeleteAccount", "Re-authentication successful: $result")
-            onSuccess()
+      reauthenticateWithGoogleSignIn(user, context, coroutineScope, onSuccess, onFailure)
+    }
+  }
+
+  private fun reauthenticateWithEmailAndPassword(
+      user: FirebaseUser,
+      password: String,
+      onSuccess: () -> Unit,
+      onFailure: () -> Unit
+  ) {
+    val email = user.email ?: throw Exception("User email is null")
+    val credential: AuthCredential = EmailAuthProvider.getCredential(email, password)
+    user.reauthenticate(credential).addOnCompleteListener { task ->
+      if (task.isSuccessful) {
+        Log.d("DeleteAccount", "User re-authenticated")
+        onSuccess()
+      } else {
+        onFailure()
+      }
+    }
+  }
+
+  private fun reauthenticateWithGoogleSignIn(
+      user: FirebaseUser,
+      context: Context,
+      coroutineScope: CoroutineScope,
+      onSuccess: () -> Unit,
+      onFailure: () -> Unit
+  ) {
+    getGoogleAuthCredential(
+        context,
+        coroutineScope,
+        onSuccess = { firebaseCredential ->
+          user.reauthenticate(firebaseCredential).addOnCompleteListener { task ->
+            if (task.isSuccessful) {
+              Log.d("FirebaseAuthRepository", "reauthenticate:success")
+              onSuccess()
+            } else {
+              Log.d("FirebaseAuthRepository", "reauthenticate:failure", task.exception)
```

Comment:
Nitpick: The logs success and failure logs for reauthenticate with google are completely different to the ones in reauthenticate wtih email. Could you please make them follow the same structure?